content/6_Advanced/Treaps.mdx

```diff
@@ -49,31 +49,62 @@ frequency: 2
-- If $\texttt{root} \leq x$, then both the root and the left subtree belong to $\texttt{left}$. We now consider a call to `split` on $\texttt{root.right}$ and note its results as $\texttt{left'}$ and $\texttt{right'}$. Finally, $\texttt{left}$ contains $\texttt{left'}$ and $\texttt{right} = \texttt{right'}$.
-- If $\texttt{root} > x$, then both the root and the right subtree belong to $\texttt{right}$. We now consider a call to `split` on $\texttt{root.left}$ and note its results as $\texttt{left'}$ and $\texttt{right'}$. Finally, $\texttt{right}$ contains $\texttt{right'}$ and $\texttt{left} = \texttt{left'}$.
+The `split` method takes in a pointer to the root of a treap $\texttt{root}$ and
+a value $x$, and returns two treaps denoted as $\texttt{left}$ and
+$\texttt{right}$. Like the name suggests, it splits the tree such that all nodes
+in $\texttt{left}$ have keys less than or equal to $x$ and all nodes in
+$\texttt{right}$ have keys greater than $x$.
+
+We can now implement it recursively. Let the left child of a node $n$ be denoted
+as $n.\texttt{left}$ and the right child as $n.\texttt{right}$.
+
+- If $\texttt{root} \leq x$, then both the root and the left subtree belong to
+  $\texttt{left}$. We now consider a call to `split` on $\texttt{root.right}$
+  and note its results as $\texttt{left'}$ and $\texttt{right'}$. Finally,
+  $\texttt{left}$ contains $\texttt{left'}$ and
+  $\texttt{right} = \texttt{right'}$.
+- If $\texttt{root} > x$, then both the root and the right subtree belong to
+  $\texttt{right}$. We now consider a call to `split` on $\texttt{root.left}$
+  and note its results as $\texttt{left'}$ and $\texttt{right'}$. Finally,
+  $\texttt{right}$ contains $\texttt{right'}$ and
+  $\texttt{left} = \texttt{left'}$.
 
 ## Merging
 
-The `merge` method inverts the `split` method by taking in two treaps $\texttt{left}$ and $\texttt{right}$ and returns a single treap that has the nodes of both treaps. It works under the assuption that all keys $x \in \texttt{left}$ are strictly smaller than all keys $y \in \texttt{right}$. Furthermore, we need to merge these two treaps such that the resultant treap still satisfies the max heap property. 
+The `merge` method inverts the `split` method by taking in two treaps
+$\texttt{left}$ and $\texttt{right}$ and returns a single treap that has the
+nodes of both treaps. It works under the assuption that all keys
+$x \in \texttt{left}$ are strictly smaller than all keys $y \in \texttt{right}$.
+Furthermore, we need to merge these two treaps such that the resultant treap
+still satisfies the max heap property.
+
+We root the resultant treap at the root node that has the higher priority, and
+recursively call `merge` on the other tree and the corresponding subtree of the
+chosen tree.
 
-We root the resultant treap at the root node that has the higher priority, and recursively call `merge` on the other tree and the corresponding subtree of the chosen tree. 
 ## Implementation
```

Comment:
it would be great if this implementation actually solved the focus problem. Probably best to move the current focus problem somewhere else and find something new in place of it?

Reply:
I'll move the code to a solution page, not sure what to replace it with tho. There aren't that many non-implicit treap problems

Reply:
Not sure why the commits are not syncing up, I've added the solution in a seperate PR https://github.com/cpinitiative/usaco-guide/pull/2258

Reply:
wdym by the commits aren't syncing up?